### PR TITLE
Eliminate hardcoded values + move them at the beginning of file

### PR DESCRIPTION
**Pull request by szykes**
## Description

The goal of this PR to make visible the hardcoded values in the code.

## Type of Change

- [x] Refactor

## Checklist

- [x] I have read the [contributing guidelines](../CONTRIBUTING.md)

### CODE REVIEW

**Review by panyuenlau — COMMENTED** [review]
Question: is it intentional to remove the comments for the existing const variables? If so, why?

**Comment by szykes** [comment]
> Question: is it intentional to remove the comments for the existing const variables? If so, why?

It was intentional. I think the comments were information duplications.

`// DefaultKafkaImage is the default Kafka image used when users don't specify it in KafkaClusterSpec.ClusterImage
DefaultKafkaImage = "ghcr.io/banzaicloud/kafka:2.13-3.4.1"`

vs

`// Kafka Cluster Spec
...
defaultKafkaImage = "ghcr.io/banzaicloud/kafka:2.13-3.4.1"`

**Comment by panyuenlau** [comment]
> I think the comments were information duplications.

I don't necessary agree this is true for all the const vars..., for example:
```golang
	// DefaultAnyCastPort kafka anycast port that can be used by clients for metadata queries
	DefaultAnyCastPort = 29092
	// DefaultIngressControllerTargetPort is the default container port for the ingress controller
	DefaultIngressControllerTargetPort = 29092
```
without the comments, I don't think these two vars are straightforward enough for others to understand what they are used for (assuming without looking into the implementation).

**Comment by szykes** [comment]
The commenting thing is not an easy topic.

I have kept the first code comment of your comment. In addition, I have added the "chart path" for most of the defaults.

File: api/v1beta1/kafkacluster_types.go [hunk 1]
```diff
@@ -37,31 +37,98 @@ import (
...
+	KafkaCRLabelKey  = "kafka_cr"
 	BrokerIdLabelKey = "brokerId"
 
-	// DefaultCruiseControlImage is the default CC image used when users don't specify it in CruiseControlConfig.Image
-	DefaultCruiseControlImage = "ghcr.io/banzaicloud/cruise-control:2.5.123"
+	// These are default values for API keys
 
-	// DefaultKafkaImage is the default Kafka image used when users don't specify it in KafkaClusterSpec.ClusterImage
-	DefaultKafkaImage = "ghcr.io/banzaicloud/kafka:2.13-3.4.1"
+	// General
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:46]:
```suggestion
	/* General Config */
```

File: api/v1beta1/kafkacluster_types.go [hunk 2]
```diff
@@ -37,31 +37,98 @@ import (
...
+	defaultServiceAccountName = "default"
+
+	// External Listener Config
+
+	// defaultAnyCastPort kafka anycast port that can be used by clients for metadata queries
+	defaultAnyCastPort                 = 29092
+	defaultIngressControllerTargetPort = 29092
+
+	// Envoy Config
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:55]:
```suggestion
	/* Envoy Config */
```

File: api/v1beta1/kafkacluster_types.go [hunk 3]
```diff
@@ -37,31 +37,98 @@ import (
...
-	// DefaultCruiseControlImage is the default CC image used when users don't specify it in CruiseControlConfig.Image
-	DefaultCruiseControlImage = "ghcr.io/banzaicloud/cruise-control:2.5.123"
+	// These are default values for API keys
 
-	// DefaultKafkaImage is the default Kafka image used when users don't specify it in KafkaClusterSpec.ClusterImage
-	DefaultKafkaImage = "ghcr.io/banzaicloud/kafka:2.13-3.4.1"
+	// General
+	defaultServiceAccountName = "default"
+
+	// External Listener Config
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:49]:
```suggestion
	/* External Listener Config */
```

File: api/v1beta1/kafkacluster_types.go [hunk 4]
```diff
@@ -37,31 +37,98 @@ import (
...
+	// KafkaClusterDeployment.spec.template.spec.container["envoy"].resource
+	defaultEnvoyRequestResourceCpu    = "100m"
+	defaultEnvoyRequestResourceMemory = "100Mi"
+	defaultEnvoyLimitResourceCpu      = "100m"
+	defaultEnvoyLimitResourceMemory   = "100Mi"
+
+	// KafkaClusterDeployment.spec.template.spec.container["envoy"].image
+	defaultEnvoyImage = "envoyproxy/envoy:v1.22.2"
+
+	// Broker Config
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:75]:
```suggestion
	/* Broker Config */
```

File: api/v1beta1/kafkacluster_types.go [hunk 5]
```diff
@@ -37,31 +37,98 @@ import (
...
+	// KafkaBrokerPod.spec.container["kafka"].resource
+	defaultBrokerRequestResourceCpu    = "1000m"
+	defaultBrokerRequestResourceMemory = "2Gi"
+	defaultBrokerLimitResourceCpu      = "1500m"
+	defaultBrokerLimitResourceMemory   = "3Gi"
+
+	defaultBrokerHeapOpts    = "-Xmx2G -Xms2G"
+	defaultBrokerPerfJvmOpts = "-server -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35 -XX:+ExplicitGCInvokesConcurrent -Djava.awt.headless=true -Dsun.net.inetaddr.ttl=60"
+
+	// Cruise Control Config
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:89]:
```suggestion
	/* Cruise Control Config */
```

File: api/v1beta1/kafkacluster_types.go [hunk 6]
```diff
@@ -37,31 +37,98 @@ import (
...
+	defaultCruiseControlTaskDurationMin = 5
+
+	// Kafka Cluster Spec
+	defaultKafkaClusterIngressController = "envoy"
+	defaultKafkaClusterK8sClusterDomain  = "cluster.local"
+
+	// KafkaBroker.spec.container["kafka"].image
+	defaultKafkaImage = "ghcr.io/banzaicloud/kafka:2.13-3.4.1"
+
+	// Istio Ingress Config
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:110]:
```suggestion
	/* Istio Ingress Config */
```

File: api/v1beta1/kafkacluster_types.go [hunk 7]
```diff
@@ -37,31 +37,98 @@ import (
...
+	defaultIstioIngressRequestResourceMemory = "128Mi"
+	defaultIstioIngressLimitResourceCpu      = "2000m"
+	defaultIstioIngressLimitResourceMemory   = "1024Mi"
+
+	// IstioMeshGateway.spec.deployment.replicas.count
+	// IstioMeshGateway.spec.deployment.replicas.min
+	// IstioMeshGateway.spec.deployment.replicas.max
+	defaultReplicas = 1
+
+	// Monitor Config
```
Comment (panyuenlau) [inline api/v1beta1/kafkacluster_types.go:123]:
```suggestion
	/* Monitor Config */
```

**Review by panyuenlau — CHANGES_REQUESTED** [review]
Comment 1 (possibly a nit): I think using block comments would create better visual separations between the different configuration sections. 

Comment 2: plz split this PR into two - one for the changes under the `api` module, and one for the rest of the changes because the refactoring in `pkg/resources/kafka/pod.go` requires a new api tag

**Review by szykes — COMMENTED** [review]
Use different comment type to separate blocks of const

**Review by panyuenlau — APPROVED** [review]
LGTM, thanks for the work!

**Review by Kuvesz — APPROVED** [review]
lgtm

**Pull request merged**